pkg/controller/perconaservermongodb/backup.go

```diff
@@ -540,6 +568,33 @@ func (r *ReconcilePerconaServerMongoDB) updatePITR(ctx context.Context, cr *api.
 		}
 	}
 
+	if cr.CompareVersion("1.16.0") >= 0 {
+		tl, err := pbm.GetLatestTimelinePITR(ctx)
+		if err != nil && err != backup.ErrNoOplogsForPITR {
+			return errors.Wrap(err, "get latest PITR timeline")
+		}
+		if err == nil {
+			bcp, err := r.getLatestBackup(ctx, cr)
+			if err != nil {
+				return errors.Wrap(err, "get latest backup")
+			}
+			if bcp != nil {
```

Comment:
how about an early return if bcp is nil

Reply:
In that case, I think it's better to move this logic to a separate function: https://github.com/percona/percona-server-mongodb-operator/pull/1486/commits/16f05f71ac7973d5f748dd5fed47d5751612610b. I don't like to have early returns in a long functions